### PR TITLE
Conformal predictor improvements

### PR DESCRIPTION
**Pull request by paxcema**
- Fixed issue where the lightwood data was saved to disk twice during the ICP -> pickle transition
- Added smoothing for classification tasks (benchmarks indicate it helps)
- Restored time series support, although only showing confidence for `nr_prediction = 1`, all further timesteps only return the predicted values
- Added exception handling in `_ts_reshape` because of a Pandas bug (reported [here](https://github.com/pandas-dev/pandas/issues/37486))
- Binned significance levels in classification for better performance
- Simplified regression test
- Added classifier test

NOTE: ~~this [PR](https://github.com/mindsdb/lightwood/pull/310/) to be merged first~~ Done.

### CODE REVIEW

File: mindsdb_native/libs/backends/lightwood.py [hunk 1]
```diff
@@ -86,7 +86,12 @@ def _ts_reshape(self, original_df):
         for i in range(len(df_arr)):
             for order_col in ob_arr + self.transaction.lmd['tss']['historical_columns']:
                 for ii in range(len(df_arr[i])):
-                    df_arr[i][order_col].iloc[ii] = [df_arr[i][order_col].iloc[ii]]
+                    try:
+                        df_arr[i][order_col].iloc[ii] = [df_arr[i][order_col].iloc[ii]]
+                    except:
```
Comment (maximlopin) [inline mindsdb_native/libs/backends/lightwood.py]: `except Exception` (or the exception that pandas raises)

File: mindsdb_native/libs/helpers/conformal_helpers.py [hunk 2]
```diff
@@ -4,7 +4,12 @@
 
 import pandas as pd
 import numpy as np
-import copy
+from copy import deepcopy
+
+
+def softmax(x, T=1, axis=0):
```
Comment (maximlopin) [inline mindsdb_native/libs/helpers/conformal_helpers.py]: maybe softmax implemented in pytorch can be used instead

File: tests/unit_tests/libs/helpers/test_conformal.py [hunk 3]
```diff
@@ -1,39 +1,69 @@
...
+    Sanity check. Confidence should be a valid probability value.
+    """
+    X, y = load_iris(return_X_y=True)
+    X_train, X_test, Y_train, Y_test = train_test_split(X, y, test_size=0.1, random_state=5)
+    target = 'medv'
+
+    x_tr = _df_from_xy(X_train, Y_train, target)
+    x_te = _df_from_xy(X_test, Y_test, target)
+
+    for label_type, type_name in zip([int, float, str], ['int', 'float', 'str']):
```
Comment (maximlopin) [inline tests/unit_tests/libs/helpers/test_conformal.py]: to get class name `__name__` property can be used:
```
>>> str.__name__
'str'
```

File: mindsdb_native/libs/helpers/conformal_helpers.py (hunk 2 above)
Comment (paxcema) [reply mindsdb_native/libs/helpers/conformal_helpers.py]: Yup, I agree. Initially opted for not adding PyTorch as a dependency, but now that we're removing Ludwig there's really no harm in doing so, right?

File: mindsdb_native/libs/backends/lightwood.py (hunk 1 above)
Comment (paxcema) [reply mindsdb_native/libs/backends/lightwood.py]: Can be more than one Exception type, I'll add the generic case

File: mindsdb_native/libs/helpers/conformal_helpers.py (hunk 2 above)
Comment (George3d6) [reply mindsdb_native/libs/helpers/conformal_helpers.py]: Pytorch install is painful, so don't add it to the requirements.txt

But it's already a dep of lightwood and will always be, so feel free to just import it without having it in the requirements.

**Review by maximlopin — APPROVED** [review]
